src/full/Agda/Interaction/BasicOps.hs

```diff
@@ -577,6 +581,47 @@ getConstraints' g f = liftTCM $ do
         let m = QuestionMark emptyMetaInfo{ metaNumber = Just $ fromIntegral ii } ii
         abstractToConcrete_ $ OutputForm noRange [] $ Assign m e
 
+-- | Goals and Warnings
+
+getGoals :: TCM Goals
+getGoals = do
+  -- visible metas (as-is)
+  visibleMetas <- typesOfVisibleMetas AsIs
+  -- hidden metas (unsolved implicit arguments simplified)
+  unsolvedNotOK <- not . optAllowUnsolved <$> pragmaOptions
+  hiddenMetas <- (guard unsolvedNotOK >>) <$> typesOfHiddenMetas Simplified
+  return (visibleMetas, hiddenMetas)
+
+getWarningsAndNonFatalErrors :: TCM WarningsAndNonFatalErrors
+getWarningsAndNonFatalErrors = do
+  mws <- getMaybeWarnings AllWarnings
+  return $ case filter isNotMeta <$> mws of
+    SomeWarnings ws@(_:_) -> swap $ classifyWarnings ws
+    _ -> ([], [])
+   where isNotMeta w = case tcWarning w of
+                         UnsolvedInteractionMetas{} -> False
+                         UnsolvedMetaVariables{}    -> False
+                         _                          -> True
```

Comment:
We could define `isMetaWarning` in `Agda.TypeChecking.Warnings` just like
we already have `isUnsolvedWarning` and use that instead rather than an
ad-hoc auxiliary function.